Sources/XMTPiOS/Client.swift

```diff
@@ -172,13 +233,36 @@ public final class Client: Sendable {
 			rustClient: client
 		)
 
-		return try Client(address: address, privateKeyBundleV1: v1Bundle, apiClient: apiClient)
+		let v3Client = try await initV3Client(
+			address: address,
+			options: options,
+			source: .static,
+			privateKeyBundleV1: v1Bundle
+		)
+
+		if let v3Client, let textToSign = v3Client.textToSign() {
+			guard let signingKey else {
+				throw ClientError.creationError("No v3 keys found, you must pass a SigningKey in order to enable alpha MLS features")
+			}
+
+			let signature = try await signingKey.sign(message: textToSign)
+			try await v3Client.registerIdentity(recoverableWalletSignature: signature.rawData)
+		}
+
+		let result = try Client(address: address, privateKeyBundleV1: v1Bundle, apiClient: apiClient, v3Client: v3Client)
+
+		for codec in options.codecs {
+			result.register(codec: codec)
+		}
```

Comment:
Is there a reason you added this here? Feels like if we didn't have it before than it doesn't make sense to add it now.

Reply:
It was busted before.